### ISSUE TITLE
Check against misspellings of assert etc. in unit tests

### ISSUE THREAD

**Issue by vabr-g** [PyCQA/pylint#3809]
Recently we cleaned up the following typos in mocks in unittests of our codebase:

- asert/aseert/assrt -> assert
- autospect/auto_spec -> autospec
- set_spec -> spec_set

Especially the asserts are dangerous, because a misspelled assert_called will fail silently on an autospec'ced mock. We found real bugs in production code which were masked by a misspelled assert_called.

I'd like to add a checker in pylint for these issues, because they are not specific to our codebase.

I'm prepared to implement it, if people of this project would be happy to have them.

**Comment by hippo91** [comment]
@vabr-g thanks for the proposition.
Have you got an example of a buggy code and its corresponding clean code?

**Comment by vabr-g** [comment]
Thanks for your reply. I cannot share the code verbatim for confidentiality reasons, but we had a test doing something like:

```
@mock.patch('some.class.method')
def testMethodIsCalled(self, some_class_method):
  self.tested_object.DoSomething()
  some_class_method.aseert_called_with(self.prepared_expectation)
```

The tested code in `DoSomething()` failed to call `some.class.method`, however the test did not notice, because the assert was misspelled. Calling `aseert_called_with` on the mock object did not raise any errors, because the mock object just thought that this is a method which should be provided.

**Comment by hippo91** [comment]
It clearly make sense to me. @whetter @PCManticore what do you think about it?

**Comment by hippo91** [comment]
@vabr-g it makes sense. Thanks for your post and keeping us aware of this PR.
I'll close this issue but depending of the evolution of https://bugs.python.org/issue41877 feel free to reopen it.
